presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -358,7 +358,9 @@ private void alterTable(HiveIdentity identity, String databaseName, String table
 
     private Partition fromMetastoreApiPartition(Table table, org.apache.hadoop.hive.metastore.api.Partition partition)
     {
-        if (isAvroTableWithSchemaSet(table)) {
+        // If a partition's Serde doesn't use metastore for schema (like Avro), we use table-level schema. Data publishers
+        // are responsible for making sure that table-level schema is workable for all partitions.
+        if (!isPartitionSerdesUsingMetastoreForSchema(partition)) {
```

Comment:
Can you please add a test (product test?) for partitioned tables where table and partition schema mismatches?

for example

- table is ORC, partition is Parquet
- table is ORC, partition is Avro with schema url
- table is ORC, partition is Avro with schema literal
- table is ORC, partition is CSV

Reply:
I was trying to add such a product test (using HQLs) which fails before the commit and succeeds after the commit, but failed to do so. But I think this commit logically makes sense and it has been deployed at our company for a long time. Please let me know if you have concerns.